src/Rules/Arrays/InvalidKeyInArrayDimFetchRule.php

```diff
@@ -32,12 +37,21 @@ public function processNode(Node $node, Scope $scope): array
 			return [];
 		}
 
-		$varType = $scope->getType($node->var);
-		if (count($varType->getArrays()) === 0) {
+		$dimensionType = $scope->getType($node->dim);
+		$varType = $this->ruleLevelHelper->findTypeToCheck(
+			$scope,
+			$node->var,
+			'',
+			static function (Type $varType) use ($dimensionType): bool {
+				$arrayDimType = $varType->setOffsetValueType($dimensionType, new MixedType());
+				return !($arrayDimType instanceof ErrorType);
```

Comment:
ah ok, sorry, I'll check this again. there is already the `stringOffsetAccess` levelstest, I thought this is testing this stuff already (because it was the one failing)